src/classLibrary/PluginHandler.ts

```diff
@@ -0,0 +1,128 @@
+import { CyanSafe, IFileFactory } from "./interfaces/interfaces";
+import chalk from "chalk";
+import { spawn } from "child_process";
+import path from "path";
+
+export class PluginHandler {
+    private readonly fileFactory: IFileFactory;
+
+    constructor(fileFactory: IFileFactory ) {
+        this.fileFactory = fileFactory;
+    }
+
+    async DownloadPlugins(cyanSafe: CyanSafe, folderName: string) {
+        let plugins: string[] = this.GetModules(cyanSafe, "api.cyanprint.dev");
+        
+        if (plugins.length > 0) {
+            if (plugins.includes("npm")) {
+                this.DownloadNpm(cyanSafe, folderName);
+            }
+            if (plugins.includes("github")) {
+                this.SetupGithub();
+            }
+        }
+    }
+
+    SetupGithub() {
+        //where to get the github details?
+        /*let git: Git = docParser.GetGit()!;
+
+        if (git) {
+            await this.ExecuteCommandSimple("git", ["init"], folderName);
+            await this.ExecuteCommandSimple("git", ["config", "user.name", git.username], folderName);
+            await this.ExecuteCommandSimple("git", ["config", "user.email", git.email], folderName);
+            await this.ExecuteCommandSimple("git", ["rm", "-rf", "--cached", "."]);
+            await this.ExecuteCommandSimple("git", ["add", "."], folderName);
+            let reply = await this.ExecuteCommand("git", ["commit", "-m", '"Initial Commit~"'], "Initialized Git repository!", folderName);
+            console.log(reply);
+
+            if (git.remote != null) {
+                let useRemote: boolean = await autoInquirer.InquirePredicate("Do you want to set your remote 'origin' to " + chalk.yellowBright(git.remote) + " and immediately push to origin?");
+                if (useRemote) {
+                    await this.ExecuteCommandSimple("git", ["remote", "add", "origin", git.remote], folderName);
+                    let reply: string = await this.ExecuteCommand("git", ["push", "origin", "--all"], "Added and pushed to remote repository", folderName);
+                    console.log(reply);
+                }
+            }
+        }*/
+    }
+
+    async DownloadNpm(cyanSafe: CyanSafe, folderName: string) {
+        let paths: string[] = cyanSafe.globs.map(glob => this.fileFactory.GetAbsoluteFilePathsOfFileInDestinationPath("package.json", glob.root, glob.pattern as string, glob.ignore)).flat();
```

Comment:
There are some errors here about type casting `glob.pattern` as `string` cause `glob.pattern` is a `string[]`, is it okay if we use `string[]`?